CONTRIBUTING.md

```diff
@@ -10,4 +10,46 @@ Docker does things, you'll feel right at home.
 Otherwise, go read
 [Docker's contributions guidelines](https://github.com/docker/docker/blob/master/CONTRIBUTING.md).
 
+### Development Environment Setup
+
+Swarm is written in [the Go programming language](http://golang.org) and manages its dependencies using [Godep](http://github.com/tools/godep). Go and Godep also use `git` to fetch dependencies.
+So, you are required to install the Go compiler, and the `git` client (e.g. `apt-get install golang git` on Ubuntu).
+
+You may need to setup your path and set the `$GOPATH` and `$GOBIN` variables, e.g:
+```sh
+mkdir -p ~/projects/swarm
+cd ~/projects/swarm
+export GOPATH=$PWD
+export GOBIN=$GOPATH/bin
+```
+
+As previously mentioned, you need `godep` to manage dependencies for Swarm. The `godep` command can be installed using the following command:
+```sh
+go get github.com/tools/godep
+```
+and you will find the `godep` binary under `$GOBIN`.
+
+Then you can fetch `swarm` code to your `$GOPATH`.
+```sh
+go get -d github.com/docker/swarm
+```
+
+The final step is to prepare dependencies for `swarm` and compile it.
+```sh
+cd src/github.com/docker/swarm
+$GOBIN/godep restore
```

Comment:
FYI instead of restoring, `$GOBIN/godep go install` will use Godep's workspace without cluttering up your $GOPATH